### ISSUE TITLE
Keep getting "One or more accounts requires reauthentication" message

### ISSUE THREAD

**Issue by amillion8** [microsoft/AzureStorageExplorer#8322]
### Preflight Checklist

- [x] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [x] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [x] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.36.2

### Architecture

x64

### Storage Explorer Build Number

20241029.2

### Platform

Windows

### OS Version

Windows11

### Bug Description

Keep getting "One or more accounts requires reauthentication" message. After I go through the process, I get same message almost immediately.

{
  "name": null,
  "message": "{\"source\":{\"accountId\":\"23bd85e0-d360-425c-a381-1c2f881b6b1a.3e20ecb2-9cb0-4df1-ad7b-914e31dcdda4:666446FFA39D8AF9E3F8ADB1ACD75DE573935F2F62E51D8AFD6362BDDCA142C3\",\"tenantId\":\"3e20ecb2-9cb0-4df1-ad7b-914e31dcdda4\"},\"resourceId\":\"https://management.core.windows.net/\",\"code\":\"AADSTS70043\",\"message\":\"The refresh token has expired or is invalid due to sign-in frequency checks by conditional access. The token was issued on 2024-07-26T10:00:08.3346138Z and the maximum allowed lifetime for this request is 43200. Trace ID: 0b2ef121-c271-4294-8400-e98008c42f00 Correlation ID: 2592ebbe-21f4-49f7-8f41-12cceff7c43b Timestamp: 2024-11-22 17:24:33Z\"}"
}

### Steps to Reproduce

1. Launch Storage Explorer
2. Expand Blob Container

![Image](https://github.com/user-attachments/assets/a8957ef0-d243-44ee-b0ad-be419810f278)

### Actual Experience

![Image](https://github.com/user-attachments/assets/d8c7698d-fe84-4b53-81cb-15fc273a5262)

**Comment by tim-myth** [comment]
Check here: https://github.com/Microsoft/AzureStorageExplorer/issues/8286

**Comment by MRayermannMSFT** [comment]
@amillion8 for that error message:

> The refresh token has expired or is invalid due to sign-in frequency checks by conditional access.

Storage Explorer must ask you to re-enter your credentials. If re-entering is not working, can you try:

1. Go to Help -> Reset, choose only Authentication, and then click Reset/Ok
1. Restart Storage Explorer
1. Try signing in

If sign-in fails, please share any error messages you see, or that are behind any sort of "error details" link.

**Comment by SylvainCME** [comment]
> [@amillion8](https://github.com/amillion8) for that error message:
> 
> > The refresh token has expired or is invalid due to sign-in frequency checks by conditional access.
> 
> Storage Explorer must ask you to re-enter your credentials. If re-entering is not working, can you try:
> 
> 1. Go to Help -> Reset, choose only Authentication, and then click Reset/Ok
> 2. Restart Storage Explorer
> 3. Try signing in
> 
> If sign-in fails, please share any error messages you see, or that are behind any sort of "error details" link.

I also noticed a bug. We have to authenticate multiple times, usually twice. After the first authentication I am able to visualize the list of blob storages and right after I select a blob storage I have to reauthenticate to visualize the content of the folder.
I noticed that a few weeks back and can confirm it is still happening every time I use the storage explorer.

**Comment by amillion8** [comment]
Followed the instructions. After the restart got below error message:

No account with ID xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX exists.

Clicked to close the message, and after logging in I get the same error as before:

One or more accounts requires reauthentication

Alex
(847) 845-6940

**Comment by craxal** [comment]
Which authentication method setting do you have? Does changing the authentication method change the outcome at all?

**Comment by amillion8** [comment]
Via Azure Entra ID

Alex
(847) 845-6940

**Comment by craxal** [comment]
Sorry, I mean what is your **Application > Sign-In > Sign In With:** setting set to?

<img width="630" alt="Image" src="https://github.com/user-attachments/assets/68531642-181f-466a-80f5-f119a2a2dd39" />

**Comment by amillion8** [comment]
Ah, got it now: Default Web Browser

Alex
(847) 845-6940

**Comment by craxal** [comment]
Does changing the Sign In method resolve the issue? Are you having similar issues logging into the Azure Portal?

**Comment by amillion8** [comment]
Changing Sign In method does not resolve the issue - getting same error. I do not have same issues with Azure Portal

Alex
(847) 845-6940

**Comment by SylvainCME** [comment]
I get it's not my thread but since I asked in here I might as well finish :) 

As it was suggested to OP, I also tried another Sign In method, chosing "Authentication Broker" instead of "Web Browser". It fixed the issue of being asked to authenticate too frequently.

**Comment by craxal** [comment]
@amillion8 Just to be thorough, did you try switching the sign in method to "Authentication Broker" after updating to 1.37.0?

**Comment by amillion8** [comment]
Yes I did. Apparently, I had to go through several iterations after switching to "Authentication Broker" during which I'd get same error message, but after try 3 or 4 the errors stopped and now Storage Explorer "behaves" as expected

Alex
(847) 845-6940

**Comment by craxal** [comment]
Sounds like we can resolve this issue then. Let us know if the problem resurfaces or additional problems come up.

**Comment by yusuf-jkhan1** [comment]
Just adding, I was getting this issue a lot too, and it was causing some (random/sporadically) of the quick access accounts to show as unauthenticated when hitting refresh all. Upgraded to 1.37 still was getting the same issues. Changed to Authentication Broker and it seems to be okay so far.
